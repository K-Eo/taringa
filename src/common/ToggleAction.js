import React from 'react'
import classNames from 'classnames'

const ToggleAction = props => {
  const icon = props.isActive ? props.activeIcon : props.inactiveIcon
  const iconClass = classNames('fa', `fa-${icon}`)
  const buttonClass = classNames('btn btn-action', props.className, {
    active: props.isActive,
  })

  return (
    <button
      {...props}
      className={buttonClass}
    >
      <i className={iconClass}/>
      {props.children}
    </button>
  )
}

ToggleAction.defaultProps = {
  isActive: false
}

export default ToggleAction
