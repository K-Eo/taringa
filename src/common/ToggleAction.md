Basic Toggle Action

```js
<ToggleAction>
  Fullscreen
</ToggleAction>
```

Toggled

```js
<ToggleAction isToggled>
  Fullscreen
</ToggleAction>
```

Toggle Action with icon
```js
<ToggleAction activeIcon="compress" inactiveIcon="expand">
  Fullscreen
</ToggleAction>
```

Toggled with icon
```js
<ToggleAction isToggled activeIcon="compress" inactiveIcon="expand">
  Fullscreen
</ToggleAction>
```
