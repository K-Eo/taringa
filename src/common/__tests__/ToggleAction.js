import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import ToggleAction from '../ToggleAction'

describe('Toggle Action', () => {
  it('renders default', () => {
    const tree = renderer
      .create(<ToggleAction>Fullscreen</ToggleAction>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders toggled', () => {
    const tree = renderer
      .create(<ToggleAction isToggled>Fullscreen</ToggleAction>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with icons', () => {
    const tree = renderer
      .create(
        <ToggleAction activeIcon="compress" inactiveIcon="expand">
          Fullscreen
        </ToggleAction>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders toggled with icons', () => {
    const tree = renderer
      .create(
        <ToggleAction activeIcon="compress" inactiveIcon="expand" isToggled>
          Fullscreen
        </ToggleAction>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('handles onClick', () => {
    const mock = jest.fn()

    const toggle = shallow(<ToggleAction onClick={mock} />)

    toggle.find('button').simulate('click')
    expect(mock).toBeCalled()
  })
})
