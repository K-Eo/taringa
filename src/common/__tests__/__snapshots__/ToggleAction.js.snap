// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Action renders default 1`] = `
<button
  className="btn btn-action"
>
  
  Fullscreen
</button>
`;

exports[`Toggle Action renders toggled 1`] = `
<button
  className="btn btn-action active"
>
  
  Fullscreen
</button>
`;

exports[`Toggle Action renders toggled with icons 1`] = `
<button
  className="btn btn-action active"
>
  <i
    className="fa fa-compress"
  />
  Fullscreen
</button>
`;

exports[`Toggle Action renders with icons 1`] = `
<button
  className="btn btn-action"
>
  <i
    className="fa fa-expand"
  />
  Fullscreen
</button>
`;
